In [15]:
from ucimlrepo import fetch_ucirepo 
import pandas as pd
import numpy as np
from sklearn.impute import SimpleImputer
from sklearn.pipeline import Pipeline
from sklearn.metrics import confusion_matrix, accuracy_score
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import KFold, cross_val_score, GridSearchCV

In [31]:
from ucimlrepo import fetch_ucirepo 
  
# fetch dataset 
breast_cancer_wisconsin_diagnostic = fetch_ucirepo(id=17) 
  
# data (as pandas dataframes) 
X = breast_cancer_wisconsin_diagnostic.data.features 
y = breast_cancer_wisconsin_diagnostic.data.targets 

X.columns

Index(['radius1', 'texture1', 'perimeter1', 'area1', 'smoothness1',
       'compactness1', 'concavity1', 'concave_points1', 'symmetry1',
       'fractal_dimension1', 'radius2', 'texture2', 'perimeter2', 'area2',
       'smoothness2', 'compactness2', 'concavity2', 'concave_points2',
       'symmetry2', 'fractal_dimension2', 'radius3', 'texture3', 'perimeter3',
       'area3', 'smoothness3', 'compactness3', 'concavity3', 'concave_points3',
       'symmetry3', 'fractal_dimension3'],
      dtype='object')

In [32]:
y['Diagnosis'].value_counts()

Diagnosis
B    357
M    212
Name: count, dtype: int64

In [35]:
lbl = LabelEncoder()
y['Diagnosis'] = lbl.fit_transform(y['Diagnosis'])

C:\Users\Sanjay Sane\AppData\Local\Temp\ipykernel_30384\3714137954.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  y['Diagnosis'] = lbl.fit_transform(y['Diagnosis'])


In [37]:
y['Diagnosis'].value_counts()

Diagnosis
0    357
1    212
Name: count, dtype: int64

In [39]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=24, test_size=0.3)

In [43]:
lr = LogisticRegression()
lr.fit(X_train, y_train)
y_pred = lr.predict(X_test)

C:\Python\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1300: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
C:\Python\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [49]:
print(confusion_matrix(y_test['Diagnosis'], y_pred))

[[101   5]
 [  5  60]]


In [51]:
accuracy_score(y_test['Diagnosis'], y_pred)

0.9415204678362573

In [53]:
y['Diagnosis'].value_counts(normalize=True)

Diagnosis
0    0.627417
1    0.372583
Name: proportion, dtype: float64

In [62]:
y_test.shape

(171, 1)

Naive Rule / Baseline Model

In [55]:
y_pred = np.zeros(y_test.shape[0])

In [57]:
y_test['Diagnosis'].value_counts()

Diagnosis
0    106
1     65
Name: count, dtype: int64

In [59]:
accuracy_score(y_test['Diagnosis'], y_pred)

0.6198830409356725

Any ML Model score should be better than score of baseline model.
Here baseline(naive) model score = 0.6198830409356725 & ML Model score = 0.9415204678362573. 
Hence we can say that this ML Model can be a relevant model. 